lib/ansible/modules/cloud/amazon/aws_guardduty_ip_list.py

```diff
@@ -0,0 +1,236 @@
+author: "Aaron Smith (@slapula)"
+version_added: "2.7"
+requirements: [ 'botocore', 'boto3' ]
+options:
+  name:
+    description:
+    - A user-friendly name that is displayed in all finding generated by activity that involves
+      IP addresses included in this list.
+    required: true
+  list_type:
+    description:
+    - Whether the list should be exist or not on the detector.
+    required: true
+    choices: ['trusted', 'threat']
+  state:
+    description:
+    - Whether the list should be exist or not on the detector.
+    default: 'present'
+    choices: ['present', 'absent']
+  enabled:
+    description:
+    - Whether the list should be enabled or disabled.
+    type: bool
+    default: true
+  detector_id:
+    description:
+    - The unique ID of the detector that you want to update.
+    required: true
+  format:
+    description:
+    - The format of the file that contains the IP list.
+    choices: ['TXT', 'STIX', 'OTX_CSV', 'ALIEN_VAULT', 'PROOF_POINT', 'FIRE_EYE']
+  location:
+    description:
+    - The URI of the file that contains the IP list.
+extends_documentation_fragment:
+    - ec2
+    - aws
+'''
+
+
+EXAMPLES = r'''
+- name: create GuardDuty detector
+  guardduty_threatlist:
+    state: present
+    enabled: true
+
+- name: disable GuardDuty detector
+  guardduty_threatlist:
+    state: present
+    enabled: false
+
+- name: delete GuardDuty detector
+  guardduty_threatlist:
+    state: absent
+'''
+
+
+RETURN = r'''
+ip_list_id:
+    description: The ID of the GuardDuty IP Set or Threat Intel Set you just created or updated.
+    returned: always
+    type: string
+'''
+
+import os
+
+from ansible.module_utils.aws.core import AnsibleAWSModule
+from ansible.module_utils.aws.core import is_boto3_error_code
+
+try:
+    from botocore.exceptions import BotoCoreError, ClientError
+except ImportError:
+    pass  # handled by AnsibleAWSModule
+
+
+def ip_list_exists(client, module):
+    try:
+        if module.params.get('list_type') == 'trusted':
+            paginator = client.get_paginator('list_ip_sets')
+            sets = paginator.paginate(
+                DetectorId=module.params.get('detector_id')
+            )
+            for i in sets:
+                for s in i['IpSetIds']:
+                    ti_set = client.get_ip_set(
+                        DetectorId=module.params.get('detector_id'),
+                        IpSetId=s
+                    )
+                    if ti_set['Name'] == module.params.get('name'):
+                        return {'exists': True, 'ip_list_id': s}
+        if module.params.get('list_type') == 'threat':
+            paginator = client.get_paginator('list_threat_intel_sets')
+            sets = paginator.paginate(
+                DetectorId=module.params.get('detector_id')
+            )
+            for i in sets:
+                for s in i['ThreatIntelSetIds']:
+                    ti_set = client.get_threat_intel_set(
```

Comment:
I think these looped calls need to have retries enabled. It seems likely that with all these in close succession there would be issues with the AWS API rate limits.